### PR TITLE
make dependencies to the openstack oslo.log and oslo.serialization li…

### PR DESCRIPTION
**Pull request by akamoto**
…braries optional

### CODE REVIEW

**Comment by akamoto** [comment]
the failing travis build refers to a line I have not touched:
ERROR: InvocationError for command '/home/travis/build/infobloxopen/infoblox-client/.tox/pep8/bin/flake8 infoblox_client' (exited with code 1)

Which for me gives:
$ flake8 --version
2.0 (pep8: 1.4.6, pyflakes: 1.3.0, mccabe: 0.2.1)
$ flake8 infoblox_client/
infoblox_client/objects.py:80:20: E901 SyntaxError: invalid syntax
$ cat -n  infoblox_client/objects.py | grep -E '^    (7[789]|8[0123])'
    77
    78      def __repr__(self):
    79          data = {field: getattr(self, field)
    80                  for field in self._fields + self._shadow_fields
    81                  if hasattr(self, field) and getattr(self, field) is not None}
    82          data_str = ', '.join(
    83              "{0}=\"{1}\"".format(key, data[key]) for key in data)

So the culprit seems to be:
                for field in self._fields + self._shadow_fields
This commit does not touch this line.

**Comment by JonasKs** [comment]
Your travis fails because of bad exceptions. [Read the entire log.](https://travis-ci.org/infobloxopen/infoblox-client/jobs/499443812#L509-L514)

Change from 
```py
try:
    from oslo_log import log as logging
except:
    import logging
```
To 
```py
try:
    from oslo_log import log as logging
except ImportError:
    import logging
```
in all your files.

File: infoblox_client/connector.py [hunk 1]
```diff
@@ -25,8 +25,15 @@
 except ImportError:
     import urllib.parse as urlparse
 
-from oslo_log import log as logging
-from oslo_serialization import jsonutils
+try:
+    from oslo_log import log as logging
+except:
```
Comment (JonasKs) [inline infoblox_client/connector.py]: except ImportError:

File: infoblox_client/object_manager.py [hunk 2]
```diff
@@ -13,7 +13,10 @@
 #    License for the specific language governing permissions and limitations
 #    under the License.
 
-from oslo_log import log as logging
+try:
+    from oslo_log import log as logging
+except:
```
Comment (JonasKs) [inline infoblox_client/object_manager.py]: except ImportError:

File: infoblox_client/objects.py [hunk 3]
```diff
@@ -15,7 +15,10 @@
 
 import six
 
-from oslo_log import log as logging
+try:
+    from oslo_log import log as logging
+except:
```
Comment (JonasKs) [inline infoblox_client/objects.py]: except ImportError:

File: infoblox_client/utils.py [hunk 4]
```diff
@@ -16,8 +16,15 @@
 import netaddr
 import six
 
-from oslo_log import log as logging
-from oslo_serialization import jsonutils
+try:
+    from oslo_log import log as logging
+except:
```
Comment (JonasKs) [inline infoblox_client/utils.py]: except ImportError:

File: infoblox_client/utils.py [hunk 5]
```diff
@@ -16,8 +16,15 @@
...
+    import logging
+
+try:
+    from oslo_serialization import jsonutils
+except:
```
Comment (JonasKs) [inline infoblox_client/utils.py]: except ImportError:

File: tests/test_connector.py [hunk 6]
```diff
@@ -18,7 +18,10 @@
 from mock import patch
 from requests import exceptions as req_exc
 
-from oslo_serialization import jsonutils
+try:
+    from oslo_serialization import jsonutils
+except:
```
Comment (JonasKs) [inline tests/test_connector.py]: except ImportError:

File: infoblox_client/connector.py (hunk 1 above)
Comment (akamoto) [reply infoblox_client/connector.py]: ah shit. Okay, thanks!

**Comment by JonasKs** [comment]
This time your codecov fails because of new code and new lines has been added, but no test to test those lines.

**Comment by JonasKs** [comment]
Reading up a bit on this, it seems if you add `# pragma: no cover` (remember to not break PEP8 while doing this) you can make sure Codecov don't read these lines.
Having tests for imports don't really makes sense, and is [not done by codecov them self](https://github.com/codecov/codecov-python/blob/master/codecov/__init__.py#L14). 

I'd change from:

```py
try:
    from oslo_log import log as logging
except ImportError:
    import logging
```
to 
```py
try:
    from oslo_log import log as logging
except ImportError:  # pragma: no cover
    import logging
``` 

This will make Codecov ignore the ImportError lines.

File: infoblox_client/connector.py [hunk 7]
```diff
@@ -25,8 +25,15 @@
...
-from oslo_log import log as logging
-from oslo_serialization import jsonutils
+try:
+    from oslo_log import log as logging
+except ImportError:
+    import logging
+
+try:
+    from oslo_serialization import jsonutils
+except ImportError:
```
Comment (JonasKs) [inline infoblox_client/connector.py]:
```py
except ImportError:  # pragma: no cover
```
Remember 2 spaces between the `:` and the ` #` to not break PEP8.

File: infoblox_client/connector.py [hunk 8]
```diff
@@ -25,8 +25,15 @@
 except ImportError:
     import urllib.parse as urlparse
 
-from oslo_log import log as logging
-from oslo_serialization import jsonutils
+try:
+    from oslo_log import log as logging
+except ImportError:
```
Comment (JonasKs) [inline infoblox_client/connector.py]: Missing 
```py
except ImportError:  # pragma: no cover
```

Comment (akamoto) [reply infoblox_client/connector.py]: thank you very much for your patience
commit cd0f4b1f9d018a6a98c103370761c05a4295ad26 (HEAD -> master, origin/master, origin/HEAD)
Author: Markus Mueller <markus.mueller@wirecard.com>
Date:   Wed Feb 27 22:08:50 2019 +0100

     add codecov no-pragma for import excepts - also when needed twice

**Comment by saiprasannasastry** [comment]
LGTM

**Pull request merged**